server/service/src/goods_received_line/query.rs

```diff
@@ -0,0 +1,41 @@
+use crate::{get_pagination_or_default, i64_to_u32, service_provider::ServiceContext, ListResult};
+
+use repository::{
+    EqualFilter, GoodsReceivedLine, GoodsReceivedLineFilter, GoodsReceivedLineRepository,
+    GoodsReceivedLineSort, PaginationOption, RepositoryError,
+};
+
+use crate::ListError;
+
+pub fn get_goods_received_lines(
+    ctx: &ServiceContext,
+    goods_received_id_option: Option<&str>,
```

Comment:
I think it would be more consistent to just use the filter here? What should the behaviour be if you have a goods_received_id of `A` but in the filter we have goods received `B`?
What you have currently is a bit ambiguous, better to just stick with the one filter I think?